### PR TITLE
wireguard: add initial support for WireGuard VPNs

### PR DESCRIPTION
**Pull request by chewitt**
WireGuard has been split into `wireguard-kernel` which tracks net-next following the initial driver merge for Linux 5.6, a new `wireless-compat-kernel` git repo for pre-5.6 kernel devices, and a new `wireless-tools` repo for the common userspace components. Linux 5.6 is on the visible horizon but still some way off so this PR adds the compat module and wireless tools as packages. Thanks are due to ConnMan maintainer Daniel Wagner for his efforts; it's now feature complete for all the configuration items needed in normal "client" use-cases. ConnMan will spawn a release sometime soon, so the githash versioning in master won't be for long.

NB: I've posted an initial wiki guide here: https://libreelec.wiki/wireguard

### CODE REVIEW

File: packages/network/libmnl/package.mk [hunk 1]
```diff
@@ -9,5 +9,6 @@ PKG_SITE="http://netfilter.org/projects/libmnl"
 PKG_URL="http://netfilter.org/projects/libmnl/files/$PKG_NAME-$PKG_VERSION.tar.bz2"
 PKG_DEPENDS_TARGET="toolchain"
 PKG_LONGDESC="A minimalistic user-space library oriented to Netlink developers."
+PKG_BUILD_FLAGS="+pic"
 
-PKG_CONFIGURE_OPTS_TARGET="--disable-shared --enable-static"
+PKG_CONFIGURE_OPTS_TARGET="--enable-shared --disable-static"
```
Comment (MilhouseVH) [inline packages/network/libmnl/package.mk]: `--disable-static --enable-shared` is the default (set by `scripts/build`) so you could drop this last line entirely to get the behaviour you want...

Comment (chewitt) [reply packages/network/libmnl/package.mk]: tested removal (along with the +fpic) and all is well, so amended.

File: packages/network/connman/package.mk [hunk 2]
```diff
@@ -44,6 +44,15 @@ PKG_CONFIGURE_OPTS_TARGET="--srcdir=.. \
...
                            --disable-silent-rules"
 
+case "$WIREGUARD_SUPPORT" in
+  yes)
+    PKG_CONFIGURE_OPTS_TARGET+=" --enable-wireguard=builtin"
+    ;;
+  *)
+    PKG_CONGIGURE_OPTS_TARGET+=" --disable-wireguard"
+    ;;
+esac
```
Comment (MilhouseVH) [inline packages/network/connman/package.mk]: Using a case for a `yes`/not `yes` test is perhaps a little excessive - normally we handle this with:
```
if [ "$WIREGUARD_SUPPORT" = "yes" ]; then
  PKG_CONFIGURE_OPTS_TARGET+=" --enable-wireguard=builtin"
else
  PKG_CONGIGURE_OPTS_TARGET+=" --disable-wireguard"
fi
```
I realise there's a `case` statement below, maybe one could argue it's a slightly more complicated test which justifies the use of `case`, but we don't need to use `case` for `yes`/not `yes` (and shouldn't, for reasons of consistency).

Comment (chewitt) [reply packages/network/connman/package.mk]: I was using an if block originally but changed it to case to match the one below. No issue to change back .. done.

**Pull request merged**